docs/examples/search_vector_similarity_examples.ipynb

```diff
@@ -19,39 +23,126 @@
    "source": [
     "import redis\n",
-    "from redis.commands.search.field import VectorField\n",
+    "\n",
+    "from redis.commands.search.indexDefinition import (\n",
+    "    IndexDefinition,\n",
+    "    IndexType\n",
+    ")\n",
     "from redis.commands.search.query import Query\n",
+    "from redis.commands.search.field import (\n",
+    "    TagField,\n",
+    "    VectorField\n",
+    ")\n",
+    "\n",
+    "r = redis.Redis(host=\"localhost\", port=6379)\n",
+    "\n",
+    "INDEX_NAME = \"index\"                       # Vector Index Name\n",
+    "VECTOR_DIMENSIONS = 1536                   # Number of Vector Dimensions\n",
+    "DOC_PREFIX = \"doc:\"                        # RediSearch Key Prefix for the Index\n",
     "\n",
-    "r = redis.Redis(host='localhost', port=36379)\n",
+    "# Schema\n",
+    "schema = (\n",
+    "    TagField(\"tag\"),\n",
+    "    VectorField(\"vector\",                  # Vector Field Name\n",
+    "        \"FLAT\", {                          # Vector Index Type: FLAT or HNSW\n",
+    "            \"TYPE\": \"FLOAT32\",             # FLOAT32 or FLOAT64\n",
+    "            \"DIM\": VECTOR_DIMENSIONS,      # Number of Vector Dimensions\n",
+    "            \"DISTANCE_METRIC\": \"COSINE\",   # Vector Search Distance Metric\n",
+    "            \"INITIAL_CAP\": 1000            # Helps with memory planning and allocation\n",
+    "        }\n",
+    "    ),\n",
+    ")\n",
+    "# NOTE: sometimes users experience issues with the INITIAL_CAP param.\n",
+    "# Best tip is to start with a smaller INITIAL_CAP value than required, especially if Redis throws an error.\n",
     "\n",
-    "schema = (VectorField(\"v\", \"HNSW\", {\"TYPE\": \"FLOAT32\", \"DIM\": 2, \"DISTANCE_METRIC\": \"L2\"}),)\n",
-    "r.ft().create_index(schema)"
+    "# Index Definition\n",
+    "definition = IndexDefinition(prefix=[DOC_PREFIX], index_type=IndexType.HASH)\n",
+    "\n",
+    "# Create Index\n",
+    "r.ft(INDEX_NAME).create_index(fields=schema, definition=definition)"
    ]
   },
   {
+   "attachments": {},
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "## Searching"
+    "## Adding Vectors to Redis\n",
+    "\n",
+    "Next, we add vectors (dummy data) to Redis using `hset`. The search index listens to keyspace notifications and will include any written HASH objects prefixed by `DOC_PREFIX`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#pip install numpy\n",
+    "import numpy as np"
    ]
   },
   {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "2"
+      ]
+     },
+     "execution_count": 4,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "r.hset(f\"doc:a\", mapping={\n",
+    "    \"vector\": np.random.rand(VECTOR_DIMENSIONS).astype(np.float32).tobytes(),\n",
```

Comment:
Consider not using random vectors here and the queries, so you can better showcase the scores as well as range queries
we usually use the L2 metric for those kinds of things in our unit tests

Reply:
I did think about this. The examples below with Cohere/OpenAI are easier to "understand" since they are real embeddings. I figured this portion would really just be focused on using the API correctly vs interpreting results. But open to a better way for sure.

Reply:
latest commit should address this